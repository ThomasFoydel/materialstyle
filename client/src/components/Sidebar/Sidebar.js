import React from 'react';
import clsx from 'clsx';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import InboxIcon from '@material-ui/icons/MoveToInbox';
import MailIcon from '@material-ui/icons/Mail';
import PropTypes from 'prop-types';

const useStyles = makeStyles({
  list: {
    width: 250,
  },
  fullList: {
    width: 'auto',
  },
});

export default function Sidebar({ props: { showSidebar, setShowSidebar } }) {
  const classes = useStyles();

  const list = (anchor) => (
    <div
      className={clsx(classes.list, {
        [classes.fullList]: anchor === 'top' || anchor === 'bottom',
      })}
      role='presentation'
      onClick={() => setShowSidebar(false)}
      onKeyDown={() => setShowSidebar(false)}
    >
      <List>
        {['Inbox', 'Starred', 'Send email', 'Drafts'].map((text, index) => (
          <ListItem button key={text}>
            <ListItemIcon>
              {index % 2 === 0 ? <InboxIcon /> : <MailIcon />}
            </ListItemIcon>
            <ListItemText primary={text} />
          </ListItem>
        ))}
      </List>
      <Divider />
      <List>
        {['All mail', 'Trash', 'Spam'].map((text, index) => (
          <ListItem button key={text}>
            <ListItemIcon>
              {index % 2 === 0 ? <InboxIcon /> : <MailIcon />}
            </ListItemIcon>
            <ListItemText primary={text} />
          </ListItem>
        ))}
      </List>
    </div>
  );

  return (
    <div>
      <Drawer
        anchor={'left'}
        open={showSidebar}
        onClose={() => setShowSidebar(false)}
      >
        {list('left')}
      </Drawer>
    </div>
  );
}

Sidebar.propTypes = {
  props: PropTypes.shape({
    showSidebar: PropTypes.bool.isRequired,
    setShowSidebar: PropTypes.func.isRequired,
  }),
};
