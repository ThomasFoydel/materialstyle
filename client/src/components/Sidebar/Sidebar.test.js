import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect,
}));

configure({ adapter: new Adapter() });

const defaultProps = {
  showSidebar: false,
  setShowSidebar: () => {},
  classes: {},
};

describe('<Sidebar />', () => {
  const container = shallow(<Sidebar props={{ ...defaultProps }} />);
  //   it('should match the snapshot', () => {
  //     expect(container.html()).toMatchSnapshot();
  //   });

  it('Should match snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('Should have three links', () => {
    expect(container.find('Link').length).toEqual(3);
  });
});
