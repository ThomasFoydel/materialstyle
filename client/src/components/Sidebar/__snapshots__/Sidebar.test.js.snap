// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> Should match snapshot 1`] = `"<div></div>"`;
